docs/csharp/tutorials/generate-consume-asynchronous-stream.md

```diff
@@ -0,0 +1,120 @@
+
+> [!WARNING]
+> Keep your personal access token secure. Any software with your personal access token could make GitHub API calls using your access rights.
+
+After creating the GitHub client, the code in `Main` creates a progress reporting object and a cancellation token. Once those objects are created, `Main` calls `runPagedQueryAsync` to retrieve the most recent 250 created issues. After that task has finished, the results are displayed.
+
+When you run the starter application, you can make some important observations about how this application runs.  You'll see progress reported for each page returned from GitHub. You can observe a noticeable pause before GitHub returns each new page of issues. Finally, the issues are displayed only after all 10 pages have been retrieved from GitHub.
+
+## Examine the implementation
+
+The implementation reveals why you observed the behavior discussed in the previous section. Examine the code for `runPagedQueryAsync`:
+
+[!code-csharp[RunPagedQueryStarter](../../../samples/csharp/tutorials/AsyncStreams/start/IssuePRreport/IssuePRreport/Program.cs#RunPagedQuery)]
+
+Let's concentrate on the paging algorithm and async structure of the preceding code. (You can consult the [GitHub GraphQL documentation](https://developer.github.com/v4/guides/) for details on the GitHub GraphQL API.) The `runPagedQueryAsync` method enumerates the issues from the most recent backwards to older issues. It requests 25 issues per page, and examines the `pageInfo` structure of the response to continue with the previous page. That follows GraphQL's standard paging support for multiple page responses. The response includes a `pageInfo` object that includes a `hasPreviousPages` value and a `startCursor` value used to request the previous page. The issues are in the `nodes` array. The `runPagedQueryAsync` method appends these nodes to an array that contains all the results from all pages.
+
+After retrieving and restoring a page of results, `runPagedQueryAsync` reports progress and checks for cancellation. If cancellation has been requested, `runPagedQueryAsync` throws an <xref:System.OperationCanceledException>.
+
+There are several elements in this code that can be improved. Most importantly, `runPagedQueryAsync` must allocate storage for all the issues returned. This sample stops at 250 issues because retrieving all open issues would require much more memory to store all the retrieved issues. In addition, the protocols for supporting progress and supporting cancellation make the algorithm harder to understand on its first reading. You must look for the progress class to find where progress is reported. You also have to trace the communications through the <xref:System.Threading.CancellationTokenSource?displayProperty=nameWithType> and its associated <xref:System.Threading.CancellationToken?displayProperty=nameTypeWith> to understand where cancellation is requested and where it is granted.
```

Comment:
Type names are not being listed consistently. Might be better to list without the namespace but here's how to change the query param if you want:

```suggestion
There are several elements in this code that can be improved. Most importantly, `runPagedQueryAsync` must allocate storage for all the issues returned. This sample stops at 250 issues because retrieving all open issues would require much more memory to store all the retrieved issues. In addition, the protocols for supporting progress and supporting cancellation make the algorithm harder to understand on its first reading. You must look for the progress class to find where progress is reported. You also have to trace the communications through the <xref:System.Threading.CancellationTokenSource?displayProperty=nameWithType> and its associated <xref:System.Threading.CancellationToken?displayProperty=nameWithType> to understand where cancellation is requested and where it's granted.
```